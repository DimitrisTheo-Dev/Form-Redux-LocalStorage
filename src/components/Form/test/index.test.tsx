import { render, screen } from "@testing-library/react"
import { Form } from "../index"
import { Provider } from "react-redux"
import configureStore from "redux-mock-store"

test("render form", () => {
  const initialState = { output: 10 }
  const mockStore = configureStore()
  let store = mockStore(initialState)
  render(
    <Provider store={store}>
      <Form />
    </Provider>
  )

  const username = screen.getByTestId("username-input")
  const description = screen.getByTestId("body-input")
  const submitButton = screen.getByTestId("submit-button")

  expect(username).toBeInTheDocument()
  expect(description).toBeInTheDocument()
  expect(submitButton).toBeInTheDocument()
})

describe("Form", () => {
  const initialState = { output: 10 }
  const mockStore = configureStore()
  it("should render", () => {
    let store = mockStore(initialState)
    const { asFragment } = render(
      <Provider store={store}>
        <Form />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
