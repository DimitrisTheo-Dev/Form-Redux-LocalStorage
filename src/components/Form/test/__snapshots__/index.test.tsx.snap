// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render 1`] = `
<DocumentFragment>
  <form
    class="css-vosf0u"
  >
    <input
      class="field"
      data-testid="username-input"
      name="username"
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      class="body"
      data-testid="body-input"
      name="body"
      placeholder="Message"
      type="text"
      value=""
    />
    <button
      class="css-jcjqud disabled"
      data-testid="submit-button"
      disabled=""
      type="submit"
    >
      <span
        class="css-1ybyu7c"
      >
        Send
      </span>
    </button>
  </form>
</DocumentFragment>
`;
